agent/utils/templates/template.go

```diff
@@ -113,3 +114,12 @@ func RenderDSN(dsn string, files *agentv1.TextFiles, tempDir string) (string, er
 	}
 	return dsn, nil
 }
+
+// CleanupTempDir removes the temporary directory.
+func CleanupTempDir(tempDir string, logger *logrus.Entry) {
+	if err := os.RemoveAll(tempDir); err != nil {
+		if logger != nil {
```

Comment:
it's so weird that we might not have logger in some calls. seems like in places where we don't have loggers we can simply send `logrus.WithField("component", "action_name")`

Reply:
Done.